src/lib.rs

```diff
@@ -212,6 +214,45 @@ impl FromStr for Message {
     }
 }
 
+impl Serialize for Message {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(self.to_string().as_str())
+    }
+}
+
+struct MessageVisitor;
+
+impl<'de> Visitor<'de> for MessageVisitor {
+    type Value = Message;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("An EIP-4361 message")
+    }
+
+    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        let message = match Message::from_str(value) {
+            Ok(message) => Ok(message),
+            Err(_error) => Err(E::custom("Error parsing message")),
```

Comment:
I believe the error should be propagated.

Reply:
It has to be wrapped there at least from what I ran into with the compiler and the serde docs, would be useful to pass through the actually error though.

Reply:
You can do something like:
```suggestion
            Err(error) => Err(E::custom(format!("error parsing message: {error}"))),
```

Reply:
iirc the error didn't implement the `display` trait when I tried to do that. Will take another pass at it.

Reply:
Nope, seems good.